pyccel/codegen/printing/fcode.py

```diff
@@ -705,7 +853,21 @@ def _print_NumpyRandint(self, expr):
         if expr.rank != 0:
             errors.report(FORTRAN_ALLOCATABLE_IN_EXPRESSION,
                           symbol=expr, severity='fatal')
-        return expr.fprint(self._print)
+        if expr.high is None:
+            randreal = self._print(PyccelMul(expr.low, NumpyRand()))
+        else:
+            randreal = self._print(PyccelAdd(PyccelMul(PyccelMinus(expr.high, expr.low), NumpyRand()), expr.low))
+
+        prec_code = self._print(iso_c_binding["integer"][expr.precision])
+        return 'floor({}, kind={})'.format(randreal, prec_code)
+
+    def _print_NumpyFull(self, expr):
+
+        # Create statement for initialization
+        if expr.fill_value is not None:
+            init_value = self._print(expr.fill_value)
+
+        return init_value
```

Comment:
If `expr.fill_value is None` then `init_value` has not been initialised and this function call will crash

Reply:
If we never expect `expr.fill_value` to be `None` then the `if` is unnecessary